pkgs/development/guile-modules/guile-sdl2/default.nix

```diff
@@ -1,46 +1,38 @@
-{ stdenv,
-  SDL2,
-  SDL2_image,
-  SDL2_ttf,
-  SDL2_mixer,
-  autoconf,
-  automake,
-  fetchgit,
-  guile,
-  libtool,
-  pkgconfig
-  }:
-stdenv.mkDerivation rec {
-  name = "guile-sdl2-${version}";
-  version = "0.1.0";
-  buildInputs = [ autoconf
-                  automake
-                  SDL2
-                  SDL2_image
-                  SDL2_ttf
-                  SDL2_mixer
-                  libtool
-                  guile
-                  pkgconfig ];
-  src = fetchgit {
-    url = "git://dthompson.us/guile-sdl2.git";
-    rev = "048f80ddb5c6b03b87bba199a99a6f22d911bfff";
-    sha256 = "1v7bc2bsddb46qdzq7cyzlw5i2y175kh66mbzbjky85sjfypb084";
+{ stdenv, fetchurl, guile, libtool, pkgconfig
+, SDL2, SDL2_image, SDL2_ttf, SDL2_mixer
+}:
+
+let
+  name = "${pname}-${version}";
+  pname = "guile-sdl2";
+  version = "0.2.0";
+in stdenv.mkDerivation {
+  inherit name;
+
+  src = fetchurl {
+    url = "https://files.dthompson.us/${pname}/${name}.tar.gz";
+    sha256 = "0yq9lsl17cdvj77padvpk3jcw2g6g0pck9jrchc7n2767rrc012b";
   };
-  preConfigurePhases = [ "bootstrapPhase" ];
-  bootstrapPhase = ''
-    ./bootstrap
-  '';
-  configureFlags = [ "--with-libsdl2-prefix=${SDL2}"
-                     "--with-libsdl2-image-prefix=${SDL2_image}"
-                     "--with-libsdl2-ttf-prefix=${SDL2_ttf}"
-                     "--with-libsdl2-mixer-prefix=${SDL2_mixer}"];
-  makeFlags = ["GUILE_AUTO_COMPILE=0"];
-  meta = {
+
+  buildInputs = [
+    guile libtool pkgconfig
+    SDL2 SDL2_image SDL2_ttf SDL2_mixer
+  ];
+
+  configureFlags = [
+    "--with-libsdl2-prefix=${SDL2}"
+    "--with-libsdl2-image-prefix=${SDL2_image}"
+    "--with-libsdl2-ttf-prefix=${SDL2_ttf}"
+    "--with-libsdl2-mixer-prefix=${SDL2_mixer}"
+  ];
+
+  makeFlags = [ "GUILE_AUTO_COMPILE=0" ];
```

Comment:
This does not appear to do anything.

Reply:
It just gets rid of compilation warnings/errors such as (WARNING, ERROR, etc):

```
;;; note: auto-compilation is enabled, set GUILE_AUTO_COMPILE=0
;;;       or pass the --no-auto-compile argument to disable.
;;; compiling /nix/store/1djqc42mqb1hacj40228cm2wq50i8c1d-guile-2.2.0/bin/guild
;;; WARNING: compilation of /nix/store/1djqc42mqb1hacj40228cm2wq50i8c1d-guile-2.2.0/bin/guild failed:
;;; ERROR: failed to create path for auto-compiled file "/nix/store/1djqc42mqb1hacj40228cm2wq50i8c1d-guile-2.2.0/bin/guild"
  GEN      sdl2/config.go
wrote `sdl2/config.go'
;;; note: auto-compilation is enabled, set GUILE_AUTO_COMPILE=0
;;;       or pass the --no-auto-compile argument to disable.
;;; compiling /nix/store/1djqc42mqb1hacj40228cm2wq50i8c1d-guile-2.2.0/bin/guild
;;; WARNING: compilation of /nix/store/1djqc42mqb1hacj40228cm2wq50i8c1d-guile-2.2.0/bin/guild failed:
;;; ERROR: failed to create path for auto-compiled file "/nix/store/1djqc42mqb1hacj40228cm2wq50i8c1d-guile-2.2.0/bin/guild"
```

I can still remove it though if you'd like?

Reply:
Like it prevents Guile from doing unnecessary things so it's useful to an extent, in my opinion, but idm not having it either.

Reply:
You are right, I did not notice this because there were no warnings when building in a terminal.

Reply:
Oh I was building in a terminal too (not emacs or anything if that's what you mean?), strange. :confused: